### ISSUE TITLE
Enum binary ops sometimes fail to generate warnings

### ISSUE THREAD

**Issue by bvdberg** [c2lang/c2compiler#542]
Also need to determine exact desired behaviour ..

```
type Enum enum u8 {
    A, B, C
}

public fn i32 main() {
    Enum e = B;
    e <<= 9;    // should give an error
    return 0;
}
```
